app/src/test/java/com/codesoom/assignment/application/UserServiceTest.java

```diff
@@ -0,0 +1,201 @@
+                        .email(EMAIL)
+                        .password(PASSWORD)
+                        .build();
+            }
+
+            @Test
+            @DisplayName("주어진 사용자를 저장하고 사용자를 리턴한다")
+            void it_returns_created_user() {
+                given(userRepository.save(any(User.class))).willReturn(user);
+
+                User user = userService.createUser(userData);
+
+                assertThat(user.getName()).isEqualTo(userData.getName());
+                assertThat(user.getEmail()).isEqualTo(userData.getEmail());
+                assertThat(user.getPassword()).isEqualTo(userData.getPassword());
+
+                verify(userRepository).save(any(User.class));
+            }
+        }
+
+        @Nested
+        @DisplayName("유효하지 않은 email을 지닌 사용자가 주어지면")
+        class Context_with_invalid_email_user {
+            private UserData invalidUserData;
+
+            @BeforeEach
+            void setUp() {
+                invalidUserData = UserData.builder()
+                        .name(NAME)
+                        .email(INVALID_EMAIL)
+                        .password(PASSWORD)
+                        .build();
+            }
+
+            @Test
+            @DisplayName("InvalidEmailException을 던진다")
+            void it_throws_invalid_email_exception() {
+                assertThatThrownBy(() -> userService.createUser(invalidUserData))
+                        .isInstanceOf(InvalidEmailException.class);
+            }
+        }
+
+        @Nested
+        @DisplayName("이미 저장된 사용자의 email이 주어지면")
+        class Context_with_stored_user {
+            private UserData stored_user_data = UserData.builder()
+                    .name(NAME)
+                    .email(EMAIL)
+                    .password(PASSWORD)
+                    .build();
+
+            @BeforeEach
+            void setUp() {
+                given(userRepository.isExistsEmail(stored_user_data.getEmail())).willReturn(true);
+            }
+
+            @Test
+            @DisplayName("DuplicateUserException을 던진다")
+            void it_throws_duplicate_user_exception() {
+                assertThatThrownBy(() -> userService.createUser(stored_user_data))
+                        .isInstanceOf(DuplicateUserException.class);
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("updateUser")
+    class Describe_updateUser {
+        private UserData updateUserData;
+
+        @BeforeEach
+        void setUp() {
+            updateUserData = UserData.builder()
+                    .name(UPDATE_NAME)
+                    .email(UPDATE_EMAIL)
+                    .password(UPDATE_PASSWORD)
+                    .build();
+        }
+
+        @Nested
+        @DisplayName("저장된 사용자 id가 주어지면")
+        class Context_with_an_stored_user_id {
+
+            @Test
+            @DisplayName("업데이트된 사용자를 리턴한다")
+            void it_returns_updated_user() {
+                User updateUser = userService.updateUser(STORED_ID, updateUserData);
```

Comment:
```suggestion
            void it_returns_updated_user() {
                given(userRepository.findById(STORED_ID)).willReturn(Optional.of(user));
                User updateUser = userService.updateUser(STORED_ID, updateUserData);
```

UserRepository를 모킹하셨었네요! userRepository가 원하는 방식으로 작동하도록 해줘야해요!

Reply:
헉!! userRepository가 원하는 방식이 무엇인지 구체적으로 조금만 더 힌트를 주시면 안 될까요.. 여기서 24시간 정체 중입니다..!

Reply:
헉헉!! 해결하였습니다;;

Reply:
어이쿠 제가 이제야 봤군요 ㅠㅜㅠㅜ 고생하셨습니다.

Reply:
제가 제안한방식으로 복붙하면 되지않았을까 싶은데 잘 안 되셨나보군요 ㅠㅜㅠㅜ

Reply:
헉헉헉!! 아닙니다. 고이 써주신 코드를 보지도 못하고 `userRepository가 원하는 방식` 코멘트에 대해서만 한참 고민하다가 삽질을 하였습니다..ㅎㅎ; 

알려주셔서 감사드립니다!